nifi-nar-bundles/nifi-airtable-bundle/nifi-airtable-processors/src/main/java/org/apache/nifi/processors/airtable/QueryAirtableTable.java

```diff
@@ -0,0 +1,422 @@
+        }
+
+        if (totalRecordCount == 0) {
+            return;
+        }
+
+        final Map<String, String> newState = new HashMap<>(state.toMap());
+        newState.put(LAST_RECORD_FETCH_TIME, currentRecordFetchDateTime);
+        try {
+            context.getStateManager().setState(newState, Scope.CLUSTER);
+        } catch (IOException e) {
+            throw new ProcessException("Failed to update cluster state", e);
+        }
+    }
+
+    private AirtableGetRecordsParameters buildGetRecordsParameters(final ProcessContext context,
+            final String lastRecordFetchTime,
+            final String nowDateTimeString) {
+        final String fieldsProperty = context.getProperty(FIELDS).evaluateAttributeExpressions().getValue();
+        final String customFilter = context.getProperty(CUSTOM_FILTER).evaluateAttributeExpressions().getValue();
+        final Integer pageSize = context.getProperty(PAGE_SIZE).evaluateAttributeExpressions().asInteger();
+
+        final AirtableGetRecordsParameters.Builder getRecordsParametersBuilder = new AirtableGetRecordsParameters.Builder();
+        if (lastRecordFetchTime != null) {
+            getRecordsParametersBuilder
+                    .modifiedAfter(lastRecordFetchTime)
+                    .modifiedBefore(nowDateTimeString);
+        }
+        if (fieldsProperty != null) {
+            getRecordsParametersBuilder.fields(Arrays.stream(fieldsProperty.split(",")).map(String::trim).collect(Collectors.toList()));
+        }
+        getRecordsParametersBuilder.customFilter(customFilter);
+        if (pageSize != null && pageSize > 0) {
+            getRecordsParametersBuilder.pageSize(pageSize);
+        }
+
+        return getRecordsParametersBuilder.build();
+    }
+
+    private void fragmentFlowFiles(final ProcessSession session, final List<FlowFile> flowFiles) {
+        final String fragmentIdentifier = UUID.randomUUID().toString();
+        for (int i = 0; i < flowFiles.size(); i++) {
+            final Map<String, String> fragmentAttributes = new HashMap<>();
+            fragmentAttributes.put(FRAGMENT_ID.key(), fragmentIdentifier);
+            fragmentAttributes.put(FRAGMENT_INDEX.key(), String.valueOf(i));
+            fragmentAttributes.put(FRAGMENT_COUNT.key(), String.valueOf(flowFiles.size()));
+
+            flowFiles.set(i, session.putAllAttributes(flowFiles.get(i), fragmentAttributes));
+        }
+    }
+
+    private void transferFlowFiles(final List<FlowFile> flowFiles, final ProcessSession session, final int totalRecordCount) {
+        session.transfer(flowFiles, REL_SUCCESS);
+        session.adjustCounter("Records Processed", totalRecordCount, false);
+        final String flowFilesAsString = flowFiles.stream().map(FlowFile::toString).collect(Collectors.joining(", ", "[", "]"));
+        getLogger().info("Successfully written {} records for flow files {}", totalRecordCount, flowFilesAsString);
```

Comment:
Recommend adjusting the wording:
```suggestion
        getLogger().info("Transferred FlowFiles [{}] Records [{}]", flowFilesAsString, totalRecordCount);
```